include/RAJA/pattern/kernel/internal/LoopTypes.hpp

```diff
@@ -0,0 +1,125 @@
+/*!
+ ******************************************************************************
+ *
+ * \file
+ *
+ * \brief   Header file for loop kernel internals: LoopData structure and
+ *          related helper functions.
+ *
+ ******************************************************************************
+ */
+
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#ifndef RAJA_pattern_kernel_internal_LoopTypes_HPP
+#define RAJA_pattern_kernel_internal_LoopTypes_HPP
+
+#include "RAJA/config.hpp"
+
+#include "camp/camp.hpp"
+
+
+namespace RAJA
+{
+namespace internal
+{
+
+namespace detail
+{
+// Helper class to convert a camp::idx_t into some type T
+// used in template expansion in ListOfNHelper
+template <typename T, camp::idx_t>
+struct SeqToType
+{
+  using type = T;
+};
+
+template <typename T, typename SEQ>
+struct ListOfNHelper;
+
+template <typename T, camp::idx_t ... SEQ>
+struct ListOfNHelper<T, camp::idx_seq<SEQ...> >
+{
+  using type = camp::list<typename SeqToType<T, SEQ>::type...>;
+};
+} // namespace detail
+
+/*
+ *  This creates a camp::list with N types, each one being T.
+ *
+ *  That is, list_of_n<T, 4>  ==  camp::list<T, T, T, T>
+ *
+ */
+template <typename T, camp::idx_t N>
+using list_of_n = typename detail::ListOfNHelper<T, camp::make_idx_seq_t<N>>::type;
```

Comment:
This seems like a thing we should probably provide in camp, perhaps `repeat<N>`?  No need to do anything with that here, but I'll add it to my list.

Reply:
That would be great: I was thinking this when I wrote it... so I'm glad you agree

Reply:
👍